netebpfext/net_ebpf_ext_xdp.c

```diff
@@ -562,6 +565,8 @@ _net_ebpf_ext_handle_xdp_tx(
 
     if (status != STATUS_SUCCESS) {
         NET_EBPF_EXT_LOG_NTSTATUS_API_FAILURE(NET_EBPF_EXT_TRACELOG_KEYWORD_XDP, "FwpsInjectMacSendAsync", status);
+        _net_ebpf_ext_l2_inject_send_complete(
+            (void*)(uintptr_t)cloned_packet, nbl, KeGetCurrentIrql() == DISPATCH_LEVEL);
```

Comment:
nit: Since `dispatch_level` is not used in `_net_ebpf_ext_l2_inject_send_complete`, do we need to call KeGetCurrentIrql() to query the current irql.

Reply:
Right, we technically do not need to. In another comment thread I posted:

> Since the dispatch level flag is therefore redundant in practice, we can fix this by either adding an IRQL check (i.e., passing the correct value) or adding one layer of indirection to the cleanup routine. Since this is an edge case and we're not micro-optimizing performance in this driver, I'll go with querying the current IRQL.

If we agree with this logic, should I add a code comment? A small refactor to avoid the need for the dispatch query is also fine.

Reply:
If the refactor is small, I think we can remove an extra call to `KeGetCurrentIrql()` as it is not needed.

Reply:
Thanks for catching the leak. Why is the dispatchLevel no longer required in the FreeClone API?

Reply:
The [documentation](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/fwpsk/nf-fwpsk-fwpsfreeclonenetbufferlist0) for FwpsFreeCloneNetBufferList says the second parameter is a flags bitmask, and all flags are publicly reserved. I imagine this routine was conflated for a different routine that takes a dispatch flag.